src/pages/workspaces/workspace/analysis/runtime-utils.test.js

```diff
@@ -495,7 +498,48 @@ describe('getDiskAppType', () => {
   })
 })
 
-describe('getCurrentPersistentDisk', () => {
+describe('getCurrentRuntime', () => {
+  it('returns undefined if no runtimes exist', () => {
+    expect(getCurrentRuntime([])).toBeUndefined()
+  })
+  it('returns a runtime if 1 exists', () => {
+    const runtime1 = getGoogleRuntime()
+    expect(getCurrentRuntime([runtime1])).toStrictEqual(runtime1)
+  })
+  it('returns no runtimes if only deleting runtimes exists', () => {
+    const runtime1 = getGoogleRuntime({ status: runtimeStatuses.deleting.label })
+    const runtime2 = getGoogleRuntime({ status: runtimeStatuses.deleting.label })
+    expect(getCurrentRuntime([runtime1, runtime2])).toBeUndefined()
+  })
+  it('returns the most recent runtime in a list', () => {
+    //chronologically, runtime1 is the middle, runtime2 the most recent, and runtime3 the oldest
+    //getCurrentRuntime should return the most recent
+    const runtime1 = getGoogleRuntime()
+    const runtime2WithSameDate = getGoogleRuntime()
+    const runtime3WithSameDate = getGoogleRuntime()
+
+    const runtime2 = {
```

Comment:
Perhaps add a check of ordering returned from `getCurrentRuntime` here before you modify the dates?

Reply:
Perhaps I'm misunderstanding what you're saying, but this test is trying to confirm that runtimes are returned depending on their chronological ordering.  The `getGoogleRuntime` data generator doesn't change the date, so I don't think `getCurrentRuntime` has defined behavior on runtimes with the same exact date (which should be impossible to occur in the wild for a given user).

Reply:
I misunderstood your comments at line 515. I thought you were saying that was the original order, but then you were changing the ordering at lines 521 and 529, and checking the effect of that in line 537. But now I see that originally the second and 3rd runtime are created with the same date…. No need to change anything.